packages/@aws-cdk-testing/framework-integ/test/aws-cloudfront-origins/test/integ.function-url-origin-oac.ts

```diff
@@ -0,0 +1,33 @@
+import * as lambda from 'aws-cdk-lib/aws-lambda';
+import * as cloudfront from 'aws-cdk-lib/aws-cloudfront';
+import * as cdk from 'aws-cdk-lib';
+import { IntegTest } from '@aws-cdk/integ-tests-alpha';
+import * as origins from 'aws-cdk-lib/aws-cloudfront-origins';
+
+const app = new cdk.App();
+
+const stack = new cdk.Stack(app, 'integ-cloudfront-function-url-origin-oac');
+
+// Create the Lambda function
+const fn = new lambda.Function(stack, 'MyFunction', {
+  code: lambda.Code.fromInline('exports.handler = async () => {};'),
+  handler: 'index.handler',
+  runtime: lambda.Runtime.NODEJS_20_X,
+});
+
+// Add a Lambda Function URL
+const fnUrl = fn.addFunctionUrl({
+  authType: lambda.FunctionUrlAuthType.AWS_IAM,
+});
+
+// Create CloudFront Distribution with OAC
+new cloudfront.Distribution(stack, 'Distribution', {
+  defaultBehavior: {
+    origin: origins.FunctionUrlOrigin.withOriginAccessControl(fnUrl),
+  },
+});
+
+// Set up integration test
```

Comment:
Can we add an assertion to this test that calls the cloudfront distribution endpoint and checks the response is as expected?

Reply:
fixed.

Reply:
Thanks for adding those assertions! What I was thinking of originally was using the `httpApiCall` method to the distribution domain and verifying the response from the lambda (something like [this test](https://github.com/aws/aws-cdk/blob/main/packages/%40aws-cdk-testing/framework-integ/test/aws-cloudfront-origins/test/integ.s3-encrypted-bucket-origin-oac.ts#L36-L39))

```
    const resp = JSON.stringify({
      statusCode: 200,
      body: 'Hello!!'
    });

    const fn = new lambda.Function(this, 'MyLambda', {
      runtime: lambda.Runtime.NODEJS_20_X,
      code: lambda.Code.fromInline(`exports.handler = async (event) => {return ${resp}}`),
      handler: 'index.handler'
    });
    ...
    const call = integ.assertions.httpApiCall(`https://${distribution.distributionDomainName}`);
    call.expect(ExpectedResult.objectLike({
      status: 200,
      body: 'Hello!!',
    }));
```
Let me know if that wouldn't work for some reason.

Reply:
@gracelu0 

Your comment passed the test correctly. This will ensure its appropriateness.